expertsystem/topology.py

```diff
@@ -152,11 +154,40 @@ def get_originating_node_list(
             self.edges[edge_id].originating_node_id for edge_id in edge_ids
         ]
 
-    def verify(self) -> bool:  # pylint: disable=no-self-use
-        """Verify if the graph is connected.
+    def verify(self) -> None:
+        """Verify if there are no dangling edges or nodes."""
+        node_ids = self.nodes | {None}  # trick to ignore None
+        for edge_id, edge in self.edges.items():
+            if (
+                edge.originating_node_id is None
+                and edge.ending_node_id is None
+            ):
+                raise ValueError(
+                    f"Edge nr. {edge_id} is not connected to any node ({edge})"
+                )
+            if not {edge.originating_node_id, edge.ending_node_id} <= node_ids:
+                raise ValueError(
+                    f"{edge} (ID: {edge_id}) has non-existing node IDs.\n"
+                    f"Available node IDs: {self.nodes}"
+                )
+        self.check_isolated_nodes()
+
+    def check_isolated_nodes(self) -> None:
+        if len(self.nodes) < 2:
+            return
+        for node_id in self.nodes:
+            surrounding_nodes = self.get_surrounding_nodes(node_id)
+            surrounding_nodes.discard(node_id)
```

Comment:
would be cleaner to discard that in the `get_surrounding_nodes` function

Reply:
e4e05576